web-common/src/features/visual-metrics-editing/MetricsTable.svelte

```diff
@@ -0,0 +1,262 @@
+<script lang="ts">
+  import { onMount } from "svelte";
+  import Checkbox from "./Checkbox.svelte";
+  import MetricsTableRow from "./MetricsTableRow.svelte";
+  import { nameWidth, labelWidth, formatWidth, ROW_HEIGHT } from "./lib";
+  import { editingItem, YAMLDimension, YAMLMeasure } from "./lib";
+
+  const headers = ["Name", "Label", "SQL expression", "Format", "Description"];
+  const gutterWidth = 56;
+
+  export let type: "measures" | "dimensions";
+  export let items: Array<YAMLDimension | YAMLMeasure>;
+  export let selected: Set<number>;
+  export let editingIndex: number | null;
+  export let searchValue: string;
+  export let reorderList: (
+    initIndex: number[],
+    newIndex: number,
+    type: "measures" | "dimensions",
+  ) => void;
+  export let onDuplicate: (
+    index: number,
+    type: "measures" | "dimensions",
+  ) => void;
+  export let onDelete: (index: number, type: "measures" | "dimensions") => void;
+  export let onCheckedChange: (checked: boolean, index?: number) => void;
+  export let onEdit: (
+    index: number,
+    type: "measures" | "dimensions",
+    field?: string,
+  ) => void;
+
+  let clientWidth: HTMLTableRowElement;
+  let tbody: HTMLTableSectionElement;
+  let scroll = 0;
+  let contentRect = new DOMRectReadOnly(0, 0, 0, 0);
+  let wrapperRect = new DOMRectReadOnly(0, 0, 0, 0);
+
+  let cursorDragStart = 0;
+  let initiatingDragIndex = -1;
+  let dragging: number[] = [];
+  let dragMovement = 0;
+  let insertIndex: number = -1;
+
+  onMount(() => {
+    const cells = clientWidth.children;
+    const initialNameWidth = cells[1].getBoundingClientRect().width;
+    const initialLabelWidth = cells[2].getBoundingClientRect().width;
+    const initialFormatWidth =
+      type === "measures" ? cells[4].getBoundingClientRect().width : 0;
+
+    nameWidth.set(initialNameWidth);
+    labelWidth.set(initialLabelWidth);
+    formatWidth.set(initialFormatWidth);
```

Comment:
I might be missing it but I don't see these stores set again, so I'm wondering if the `MaxStore` layer is needed?

Reply:
They are all set once per table, so twice for each store.